src/common/mock-service.c

```diff
@@ -454,6 +456,68 @@ on_tell_me_your_name (TestFrobber *frobber,
 
 }
 
+static gboolean
+test_fd_pipe_readable (gint fd,
+                       GIOCondition condition,
+                       gpointer user_data)
+{
+  const char *expected = "Hello, fd";
+  char buffer[100];
+
+  g_assert (condition == G_IO_IN);
+
+  read (fd, buffer, 100);
```

Comment:
semaphore complains here while checking:
```
src/common/mock-service.c: In function ‘test_fd_pipe_readable’:
src/common/mock-service.c:469:8: error: ignoring return value of ‘read’, declared with attribute warn_unused_result [-Werror=unused-result]
   read (fd, buffer, 100);
        ^
```
Even in a mock function it's probably good practice to check this.